src/python/pants/backend/javascript/install_node_package.py

```diff
@@ -91,6 +96,14 @@ async def install_node_packages_for_address(
     )
     package_digest = source_files.snapshot.digest
 
+    subsystem_env_vars, target_env_vars = await MultiGet(
```

Comment:
As written I've only applied the env vars to the install command, but I could move them to either 

1. All commands that use the package manger (extend this set https://github.com/pantsbuild/pants/blob/b8070ff9bcf8c79dfc862808406fffdba4266f19/src/python/pants/backend/javascript/package_manager.py#L78)
2. A specific subset of the commands 

The original issue @kevin-oliveira-zocdoc opened was about `YARN_NODE_LINKER` which would need to be on every invocation (install, run, pack, etc) So I could move this setup_nodejs_project_environment_process` which I think would work

Reply:
I've moved it into `setup_nodejs_project_environment_process` which will affect all usages of `NodeJsProjectEnvironmentProcess` (This is all package manager operations (install/generate-lockfiles, but also all nodejs operations, check/test/lint because they're run via the package manager)

Reviewing the existing goals for existing env var options

Test: This goal uses the `TestExtraEnvVarsField` on the target and env vars on Test Subsystem 
Run & Package `node_build_script`:  `extra_env_vars` field
Tools (prettier, openapi): no existing env vars, but the property is available on NodeJSToolBase so could be used in a custom plugin rule

Maybe the subsystem option name should be broadened? `[nodejs].extra_env_vars`?

Reply:
Thanks

> Maybe the subsystem option name should be broadened? [nodejs].extra_env_vars?

Ah yeah, that sounds good too.

Reply:
updated